src/ui/favoritespanewidget.cc

```diff
@@ -1000,6 +1000,7 @@ TreeItem * FavoritesModel::getItemByFullPath( const QStringList & fullPath ) con
         if ( item->data().toString() == fullPath[ fullPathPos ] ) {
           if ( fullPathPos == ( fullPath.count() - 1 ) ) {
             targetFolder = item;
+            return targetFolder;
```

Comment:
not work as expected , when code reach this line ,it must already in the last run.

Reply:
the implementation is wrong ,just as long as the last path(in the fullPath) matched ,it will return.